src/embed-player.js

```diff
@@ -388,21 +388,28 @@ export default class EmbedPlayer {
     }
 
     setupChromecast(selector, chromecastReceiverAppId) {
-        const castButtonContaner = document.querySelector(selector);
-        const castButton = document.createElement('google-cast-launcher');
-        castButtonContaner.appendChild(castButton);
-        if (chromecastReceiverAppId) {
-            window['__onGCastApiAvailable'] = (isAvailable) => {
-                if (isAvailable && cast && cast.framework) {
-                    this.initializeCastApi(chromecastReceiverAppId);
-                }
-            };
+        return new Promise(resolve => {
+            const castButtonContaner = document.querySelector(selector);
+            const castButton = document.createElement('google-cast-launcher');
+            castButtonContaner.appendChild(castButton);
+            if (chromecastReceiverAppId) {
+                window['__onGCastApiAvailable'] = (isAvailable) => {
+                    if (isAvailable && cast && cast.framework) {
+                        this.initializeCastApi(chromecastReceiverAppId);
```

Comment:
should the promise be resolved from here?

Reply:
done